### PR TITLE
js/melpa.js: allow https://git.* fallback for other forges

### PR DESCRIPTION
**Pull request by stsquad**
The calculateSourceURL function has special handling for a selection
of forges and other hosting sites. Rather than adding new special
handling for forges like sr.ht lets just assume that a modern https://
website starting with git with do the right thing when a browser
visits.

### CODE REVIEW

File: html/js/melpa.js [hunk 1]
```diff
@@ -136,7 +136,8 @@
                 urlMatch(/(gitlab\.com\/[^\/]+\/[^.]+)/, "https://") ||
                 urlMatch(/^lp:(.*)/, "https://launchpad.net/") ||
                 urlMatch(/^(https?:\/\/code\.google\.com\/p\/[^\/]+\/)/) ||
-                urlMatch(/^(https?:\/\/[^.]+\.googlecode\.com\/)/));
+                urlMatch(/^(https?:\/\/[^.]+\.googlecode\.com\/)/) ||
+                urlMatch(/^(https?:\/\/git.*)/));
```
Comment (mattiasb) [inline html/js/melpa.js]: Just a drive-by comment:
`/^(https?:\/\/git.*)/` should probably be `/^(https?:\/\/git\.)/`

**Comment by purcell** [comment]
Looks good, but I agree that the fix noted by @mattiasb appears necessary here, otherwise the regex itself will incorrectly match `gitlab`, `github` etc. (though of course there are separate earlier matchers for those).

**Comment by purcell** [comment]
P.S. did you test-drive this locally?

**Comment by purcell** [comment]
Modified version committed, thanks.
